src/middlewared/middlewared/plugins/pool.py

```diff
@@ -208,6 +208,14 @@ class PoolService(CRUDService):
         options['prefix'] = 'vol_'
         return await self.middleware.call('datastore.query', 'storage.volume', filters, options)
 
+    @accepts()
+    async def file_system_choices(self, ):
```

Comment:
there is an extra comma

better we call it filesystem_choices.

I think you can reimplement list_zfs_fsvols using zfs.dataset.query which uses libzfs.

We dont want to port poorly written code from notifier.py to zfs plugin using CLI calls.